package/index.ts

```diff
@@ -4466,6 +4466,12 @@ function doAction(transition: IRouteTransition) {
 declare var Promise: any;
 
 function nextIteration(): void {
+    if (lastRunnedTransitionInfo.transition && lastRunnedTransitionInfo.transition.name === currentTransition.name) {
+        lastRunnedTransitionInfo.runCount++;
+    } else {
+        lastRunnedTransitionInfo = { transition: currentTransition, runCount: 1 };
+    }
+
     while (true) {
```

Comment:
Why this complexity here, when you could just increment global counter in runTransition?